scipy/stats/_resampling.py

```diff
@@ -658,59 +671,65 @@ def _monte_carlo_test_iv(sample, rvs, statistic, vectorized, n_resamples,
     if alternative not in alternatives:
         raise ValueError(f"`alternative` must be in {alternatives}")
 
-    return (sample, rvs, statistic_vectorized, vectorized, n_resamples_int,
+    return (data_iv, rvs, statistic_vectorized, vectorized, n_resamples_int,
             batch_iv, alternative, axis_int)
 
 
 fields = ['statistic', 'pvalue', 'null_distribution']
 MonteCarloTestResult = make_dataclass("MonteCarloTestResult", fields)
 
 
-def monte_carlo_test(sample, rvs, statistic, *, vectorized=None,
+@_rename_parameter('sample', 'data')
```

Comment:
When I write "sample", I mean a group of observations. I think it's confusing enough that "a sample" is sometimes used interchangeably with "an observation".  I wouldn't want it to _also_ mean a group of groups of observations!

I see your point in the sense that a set can contain other sets, but I don't think "sample" is used that way anywhere else in the SciPy documentation. On the other hand, `data` doesn't imply one sample or many, and it is the name of the first parameters of the closely related functions `bootstrap` and `permutation_test`.

We could deprecate `sample`. Deprecation is easy to implement just by adding the deprecation version as the third argument, but this would need approval on the mailing list. I try to save my requests to deprecate something for when it really matters to me. Personally, I wouldn't mind supporting either name forever - or until someone else wants to deprecate the old name.

Reply:
I agree with changing `sample` to `data`. If each sample in data was required to have the same number of observations, one could make a good case for calling the whole thing a single sample from a multivariate distribution, but given that the samples can be of different sizes, calling the whole thing a sample seems like it would be misleading. I like the choice of `data`.